lib/ClangImporter/ImportDecl.cpp

```diff
@@ -3352,15 +3352,38 @@ namespace {
         }
       }
 
+      bool retunrsRetainedUnretainedDiagnosticsNeeded = false;
```

Comment:
I think it might be cleaner to put every logic about determining whether we want to warn in a separate function instead of having the logic split between below and a separate function (`isNormalCxxMemberMethod`).

Reply:
Agreed with Gabor's point here. Another perspective is that assigning `retunrsRetainedUnretainedDiagnosticsNeeded` here is doing unnecessary work in the common/happy case where no diagnostics are needed at all. See my comment below.

Reply:
Just to clarify: this conversation is being addressed in https://github.com/swiftlang/swift/pull/77397, right?